WalletWasabi/WabiSabi/Client/Decomposer.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Client;
+
+/// <summary>
+/// https://github.com/lontivero/DecompositionsPlayground/blob/master/Notebook.ipynb
+/// </summary>
+public static class Decomposer
+{
+	public static long[] StdDenoms;
+
+	public static IEnumerable<(long Sum, int Count, ulong Decomposition)> Decompose(long target, long tolerance, int maxCount)
+	{
+		if (maxCount is <= 1 or > 8)
+		{
+			throw new ArgumentOutOfRangeException(nameof(maxCount), "The maximum decomposition lenght cannot be greater than 8 or smaller than 1.");
```

Comment:
According to `if (maxCount is <= 1 or > 8)` the message should be:

```suggestion
			throw new ArgumentOutOfRangeException(nameof(maxCount), "The maximum decomposition length cannot be greater than 8 or smaller than 2.");
```